scripts/memory_logger.py

```diff
@@ -1,73 +1,55 @@
+    for i in xrange(process.GetThreadAtIndex(0).GetNumFrames()):
+      lldb_results.append(process.GetThreadAtIndex(0).GetFrameAtIndex(i).__str__())
+    process.Detach()
+    return '\n'.join(lldb_results)
 
-  def __del__(self):
-    lldb.SBDebugger.Destroy(self.debugger)
+class GDB:
+  def __init__(self):
+    self.last_position = 0
 
-  def _parseStackTrace(self, gibberish):
-    return gibberish
-
-  def _run_commands(self, commands):
-    tmp_text = ''
-    return_obj = lldb.SBCommandReturnObject()
-    for command in commands:
-      self.command_interpreter.HandleCommand(command, return_obj)
-      if return_obj.Succeeded():
-        if command == 'process status':
-          tmp_text += '\n########################################################\n## Process Status:\n##\n'
-          tmp_text += return_obj.GetOutput()
-        elif command == 'bt':
-          tmp_text += '\n########################################################\n## Backtrace:\n##\n'
-          tmp_text += return_obj.GetOutput()
-    return tmp_text
-
-  def getStackTrace(self, pid):
-    event = lldb.SBEvent()
-    lldb_results = ''
-    state = 0
-    attach_info = lldb.SBAttachInfo(int(pid))
-    process = self.target.Attach(attach_info, self.error)
-    process.GetBroadcaster().AddListener(self.listener, lldb.SBProcess.eBroadcastBitStateChanged)
-    done = False
-    while not done:
-      if self.listener.WaitForEvent(lldb.UINT32_MAX, event):
-        state = lldb.SBProcess.GetStateFromEvent(event)
-        if state == lldb.eStateExited:
-          done = True
-        elif state == lldb.eStateStopped:
-          lldb_results = self._run_commands(['process status', 'bt', 'cont'])
-          done = True
-        elif state == lldb.eStateRunning:
-          self._run_commands(['process interrupt'])
-      if state == lldb.eStateCrashed or state == lldb.eStateInvalid or state == lldb.eStateExited:
-        return 'Binary exited before sample could be taken'
-      time.sleep(0.03)
-
-    # Due to some strange race condition we have to wait until eState is running
-    # before we can pass the 'detach, quit' command. Why we can not do this all in
-    # one go... bug?
-    done = False
-    while not done:
-      if self.listener.WaitForEvent(lldb.UINT32_MAX, event):
-        state = lldb.SBProcess.GetStateFromEvent(event)
-        if state == lldb.eStateRunning:
-          self._run_commands(['detach', 'quit'])
-          done = True
-      if state == lldb.eStateCrashed or state == lldb.eStateInvalid or state == lldb.eStateExited:
-        return 'Binary exited before sample could be taken'
-      time.sleep(0.03)
-    return self._parseStackTrace(lldb_results)
+  def __detach__(self, process):
+    process, gdb_stdout = process
+    for command in ['quit\n', 'y\n']:
+      if self._waitForResponse(gdb_stdout):
+        try:
+          process.stdin.write(command)
+        except:
+          process.send_signal(signal.SIGKILL)
```

Comment:
SIGKILL here? You have it only in the `except` branch so maybe that's OK. I'm a little confused at what this method is doing exactly. In fact inline documentation in this code is a little on the sparse side... (and by a little I mean a lot)